### PR TITLE
add JNI channel to dispatch device event

### PR DESCRIPTION
**Pull request by nicelyjust**
#### Problem
so far can not receive device event in Android，such as CommissioningComplete event
#### Change overview
add JNI channel to dispatch device event

#### Testing
- run ./scripts/examples/gn_android_example.sh examples/tv-app/android out/android-arm-chip-tvserver
- tested chip pairing and basic clusters

### CODE REVIEW

**Review by chrisdecenzo — APPROVED** [review]

**Review by amitnj — APPROVED** [review]

File: examples/tv-app/android/java/DeviceCallbacks.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    ChipLogProgress(Zcl, "TV Android App: set ChipDeviceEvent delegate");
+    DeviceCallbacks * mgr = new DeviceCallbacks();
+    PlatformMgr().AddEventHandler(OnPlatformEventWrapper, reinterpret_cast<intptr_t>(mgr));
+    mgr->InitializeWithObjects(manager);
+}
+
+void DeviceCallbacks::InitializeWithObjects(jobject provider)
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrReturn(env != nullptr, ChipLogError(Zcl, "Failed to GetEnvForCurrentThread for DeviceEventProvider"));
```
Comment (andy31415) [inline examples/tv-app/android/java/DeviceCallbacks.cpp:61]: is `Zcl` the right domain for these errors?

`DeviceLayer` ? `AppServer` ? `Support` ? maybe more suitable

File: examples/tv-app/android/java/DeviceCallbacks.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{
+
+public:
+    static void NewManager(jobject manager);
+    void OnPlatformEvent(const chip::DeviceLayer::ChipDeviceEvent * event);
+    void InitializeWithObjects(jobject manager);
+
+private:
+    jobject mProvider                     = nullptr;
+    jmethodID commissioningCompleteMethod = nullptr;
```
Comment (andy31415) [inline examples/tv-app/android/java/DeviceCallbacks.h:32]: s/commissioningCompleteMethod/mCommissioningCompleteMethod/

File: examples/tv-app/android/java/DeviceCallbacks.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        OnSessionEstablished(event);
+        break;
+    }
+}
+
+void DeviceCallbacks::OnSessionEstablished(const ChipDeviceEvent * event)
+{
+    if (event->SessionEstablished.IsCommissioner)
+    {
+        printf("Commissioner detected!");
```
Comment (andy31415) [inline examples/tv-app/android/java/DeviceCallbacks.cpp:93]: Is printf intended? Why not logging methods?

**Review by andy31415 — APPROVED** [review]

File: examples/tv-app/android/java/DeviceCallbacks.cpp (hunk 1 above)
Comment (nicelyjust) [reply examples/tv-app/android/java/DeviceCallbacks.cpp]: Thanks for the suggestion

**Pull request merged**